From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: SpiritCroc <dev@spiritcroc.de>
Date: Tue, 3 Aug 2021 12:44:52 +0200
Subject: [PATCH 076/226] Reduce NotificationHistoryDatabase logspam

If the file is not there, we do not need the whole stacktrace.

Change-Id: Ie727f678b72940f05d2c485f5ff43f4567415fb2
Signed-off-by: Dmitrii <bankersenator@gmail.com>
---
 .../notification/NotificationHistoryDatabase.java | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/services/core/java/com/android/server/notification/NotificationHistoryDatabase.java b/services/core/java/com/android/server/notification/NotificationHistoryDatabase.java
index 40c0a53553..9c74535066 100644
--- a/services/core/java/com/android/server/notification/NotificationHistoryDatabase.java
+++ b/services/core/java/com/android/server/notification/NotificationHistoryDatabase.java
@@ -178,6 +178,8 @@ public class NotificationHistoryDatabase {
                 try {
                     readLocked(
                             file, notifications, new NotificationHistoryFilter.Builder().build());
+                } catch (FileNotFoundException e) {
+                    Slog.e(TAG, "error reading " + file.getBaseFile().getAbsolutePath() + ", " + e);
                 } catch (Exception e) {
                     Slog.e(TAG, "error reading " + file.getBaseFile().getAbsolutePath(), e);
                 }
@@ -204,6 +206,8 @@ public class NotificationHistoryDatabase {
                         // No need to read any more files
                         break;
                     }
+                } catch (FileNotFoundException e) {
+                    Slog.e(TAG, "error reading " + file.getBaseFile().getAbsolutePath() + ", " + e);
                 } catch (Exception e) {
                     Slog.e(TAG, "error reading " + file.getBaseFile().getAbsolutePath(), e);
                 }
@@ -307,7 +311,7 @@ public class NotificationHistoryDatabase {
             in = file.openRead();
             NotificationHistoryProtoHelper.read(in, notificationsOut, filter);
         } catch (FileNotFoundException e) {
-            Slog.e(TAG, "Cannot open " + file.getBaseFile().getAbsolutePath(), e);
+            Slog.e(TAG, "Cannot open " + file.getBaseFile().getAbsolutePath() + ", " + e);
             throw e;
         } finally {
             if (in != null) {
@@ -373,6 +377,9 @@ public class NotificationHistoryDatabase {
                                 new NotificationHistoryFilter.Builder().build());
                         notifications.removeNotificationsFromWrite(mPkg);
                         writeLocked(af, notifications);
+                    } catch (FileNotFoundException e) {
+                        Slog.e(TAG, "Cannot clean up file on pkg removal "
+                                + af.getBaseFile().getAbsolutePath() + ", " + e);
                     } catch (Exception e) {
                         Slog.e(TAG, "Cannot clean up file on pkg removal "
                                 + af.getBaseFile().getAbsolutePath(), e);
@@ -416,6 +423,9 @@ public class NotificationHistoryDatabase {
                         if(notificationHistory.removeNotificationFromWrite(mPkg, mPostedTime)) {
                             writeLocked(af, notificationHistory);
                         }
+                    } catch (FileNotFoundException e) {
+                        Slog.e(TAG, "Cannot clean up file on notification removal "
+                                + af.getBaseFile().getName() + ", " + e);
                     } catch (Exception e) {
                         Slog.e(TAG, "Cannot clean up file on notification removal "
                                 + af.getBaseFile().getName(), e);
@@ -460,6 +470,9 @@ public class NotificationHistoryDatabase {
                                 mPkg, mConversationIds)) {
                             writeLocked(af, notificationHistory);
                         }
+                    } catch (FileNotFoundException e) {
+                        Slog.e(TAG, "Cannot clean up file on conversation removal "
+                                + af.getBaseFile().getName() + ", " + e);
                     } catch (Exception e) {
                         Slog.e(TAG, "Cannot clean up file on conversation removal "
                                 + af.getBaseFile().getName(), e);
